go/ct/gen/state_test.go

```diff
@@ -478,3 +506,110 @@ func TestStateGenerator_BlockNumberHashes(t *testing.T) {
 		newHashes = append(newHashes, hashi)
 	}
 }
+
+func TestStateGenerator_VariousContraints(t *testing.T) {
```

Comment:
These tests seem to focus on the effect of the functions registering conditions on the generator -- which is, on its own fine. What I am a bit surprised / worried is that these functions have not been called as part of the test making sure that those constraints can be satisfied. Are there no such tests?

Reply:
there are such a tests, like the one in line 296, but since some conditions clash, not all of them are called. I could modify this test to generate a state and check that the condition is also met afterwards, but I feel that would also be testing the generate function, not really the constraining functions